federation/backends/minimal_profile.md

```diff
@@ -0,0 +1,194 @@
+earth observation data.
+
+This document does not aim to contradict the openEO API, and when this is perceived to be the case, the definitions in the openEO API get precedence.
+It is possible that requirements in this document do define additional constraints on top of the openEO API.
+
+This profile is established because the openEO web service API is quite liberal if interpreted strictly. So a web service can be compliant to the API 
+specification, but still be mostly incompatible with other backends, and existing usage patterns. By following this profile, implementors have a better
+view on what is important.
+
+## Scope
+This minimal profile aims to help initial openEO implementations, that seek a least effort approach to achieve minimal compatibility with other backends.
+Requirements in this profile are not sufficient for backends in the openEO platform federation.
+
+## Mandatory endpoints
+
+In the openEO API, all endpoints are optional and discoverable. These are the endpoints that are required by this profile:
+
+
+
+```
+[
+    {
+      "methods": [
+        "GET"
+      ],
+      "path": "/file_formats"
+    },
+    {
+      "methods": [
+        "GET"
+      ],
+      "path": "/collections"
+    },
+    {
+      "methods": [
+        "GET"
+      ],
+      "path": "/collections/{collection_id}"
+    },
+    {
+      "methods": [
+        "GET"
+      ],
+      "path": "/processes"
+    },
+    {
+      "methods": [
+        "GET",
+        "POST"
+      ],
+      "path": "/jobs"
+    },
+    {
+      "methods": [
+        "GET",
+        "DELETE"
+      ],
+      "path": "/jobs/{job_id}"
+    },
+    {
+      "methods": [
+        "GET",
+        "POST",
+        "DELETE"
+      ],
+      "path": "/jobs/{job_id}/results"
+    },
+    {
+      "methods": [
+        "GET"
+      ],
+      "path": "/jobs/{job_id}/results/{user_base64}/{secure_key}"
+    },
+    {
+      "methods": [
+        "GET"
+      ],
+      "path": "/jobs/{job_id}/results/items/{user_base64}/{secure_key}/{item_id}"
+    },
+    {
+      "methods": [
+        "GET"
+      ],
+      "path": "/jobs/{job_id}/results/items/{item_id}"
+    },
+    {
+      "methods": [
+        "GET"
+      ],
+      "path": "/jobs/{job_id}/results/assets/{filename}"
+    },
+    {
+      "methods": [
+        "GET"
+      ],
+      "path": "/jobs/{job_id}/results/assets/{user_base64}/{secure_key}/{filename}"
+    }
+  ],
+```
+
+## Batch job support
+This backend mandates support for batch jobs over the synchronous jobs via the /result endpoint.
+Both are very relevant, but batch jobs are not constrained in processing time.
+
+## File formats
+GeoTiff is the only mandatory output format: https://docs.openeo.cloud/federation/backends/fileformats.html#geotiff
+
+## Collections
+A minimal backend shall have at least one raster collection, allowing to build raster data cubes. 
+
+## Processes
```

Comment:
We already have such a list: https://github.com/openEOPlatform/documentation/blob/main/federation/backends/processes.md

Reply:
I know, but the problem is that ESA asked for something even more minimal, as the core profile for openEO platform is already quite demanding. (Minimal openEO backends would not be able to join the openEO platform federation, then they would indeed hace to support the core profile.)

Reply:
Hmm... somewhat makes sense. A couple of questions arise though:
- How was this list compiled? Looking at the "minimal" list is looks like it could be more minimal (e.g. remove normalized_difference? filters?).
- What's the use of having two lists? One specified a minimal profile, the other one was until now a requirement to be implemented as part of the federation contract. That's contradicting. 
- Should we have this in one place and one table?
- Do these lists actually make sense at all? Maybe more as a guideline? A "vector backend" would never be able to fullfill these...

Reply:
Imo it makes no sense to provide and maintain two different lists as platform if we don't do anything with this one here.
As it's not that easy to actually fulfil the platform core profile (as discussed our backend is missing some of them) I would be in favour of a simpler list. Having all processes listed here would of course not mean that you automatically become a platform backend (there are other points to consider that should be clearer defined somewhere).

Reply:
A profile is indeed mostly a guideline. Basically, anyone wanting to implement a backend will need guidance on which processes to implement first, and then a minimal list like this can be helpful. 
If we don't have a 'minimal' list for raster backends, providers can implement anything at all and call themselves openEO compliant.
If we only have the openEO platform list, then the threshold is quite high.
Hence the two lists?
As to the contents of the list, I'm fine with reducing it or adapting in any way.
I specified it here because didn't have another good location, but openEO.org is also a good option?

Reply:
> If we don't have a 'minimal' list for raster backends, providers can implement anything at all and call themselves openEO compliant. [...] but openEO.org is also a good option?

Indeed and that is intentional. You can choose what you need or want, openEO doesn't tell you what you have to implement. If you just implement load_collection and save_result you can be openEO compliant and I don't think why we should raise the bar. Guidance is okay, but it shouldn't be required to be openEO compliant. But that's unrelated to openEO Platform, where we can define a profile for becoming part of the federation or so. But we have two lists now and I still don't exactly understand why we need two of them.

If you actually want to define a "minimal" guideline for openEO itself, I'd say propose it in the PSC and/or community meetings. If this is meant for openEO Platform then it's a different story and should live here.

Reply:
Ok, moved the discussion elsewhere so we can indeed have it in the PSC:
https://github.com/Open-EO/openeo-api/issues/490